pkg/ttl/ttlworker/task_manager.go

```diff
@@ -597,6 +601,33 @@ func (t *runningScanTask) Context() context.Context {
 	return t.ctx
 }
 
-func (t *runningScanTask) finished() bool {
-	return t.result != nil && t.statistics.TotalRows.Load() == t.statistics.ErrorRows.Load()+t.statistics.SuccessRows.Load()
+func (t *runningScanTask) finished(logger *zap.Logger) bool {
+	if t.result == nil {
+		// Scan task isn't finished
+		return false
+	}
+
+	logger = logger.With(
+		zap.String("jobID", t.JobID),
+		zap.Int64("scanID", t.ScanID),
+		zap.String("table", t.tbl.Name.O),
+	)
+
+	if t.statistics.TotalRows.Load() <= t.statistics.ErrorRows.Load()+t.statistics.SuccessRows.Load() {
+		// All rows are processed.
+		// We use `<=` instead of `==` to make the logic strong to make sure
+		// it also works when statistics are not accurate.
+		logger.Info("mark TTL task finished because all scanned rows are processed")
+		return true
+	}
+
+	if time.Since(t.result.time) > waitTaskProcessRowsTimeout {
```

Comment:
Is two minutes long enough? Previously, the meaning of `finished()` is that both `scan` and `delete` are finished. If we are going to keep the behavior, it means the `delete` should have finished within 2 minutes, right?

Reply:
Maybe I can enlarge it to 5 minutes. The `delChan` is a channel with 0 zero size, and san worker will only be marked as finished after the `delTask` emitted. That means we as most waiting `tidb_ttl_scan_batch_size` rows to deleted. `tidb_ttl_scan_batch_size` default value is 500, I think it is enough for normal cases